### PR TITLE
CAL-1426 allow to list 100 calendars available in event creation form

### PR DESCRIPTION
**Pull request by trongtt**

### CODE REVIEW

File: calendar-webservice/src/main/java/org/exoplatform/calendar/ws/CalendarRestApi.java [hunk 1]
```diff
@@ -3796,10 +3796,18 @@ private Response buildBadResponse(ErrorResource error) {
...
-   * Doesn't allow limit parameter to exceed the default query_limit
+   * Use default value if limit is not set. And do not allow to exceed the hard limit 100.
    */
   private int parseLimit(int limit) {
-    return (limit <=0 || limit > query_limit) ? query_limit : limit;
+    if (limit <= 0) {
+      return query_limit;
+    }
+
+    if (limit > 100) {
```
Comment (thomasdelhomenie) [inline calendar-webservice/src/main/java/org/exoplatform/calendar/ws/CalendarRestApi.java]: 100 should be defined in a class member, like query_limit.
Also, as you proposed, we can make it configurable like query_limit.

File: calendar-webservice/src/main/java/org/exoplatform/calendar/ws/CalendarRestApi.java [hunk 2]
```diff
@@ -3796,10 +3796,18 @@ private Response buildBadResponse(ErrorResource error) {
   }
 
   /**
-   * Doesn't allow limit parameter to exceed the default query_limit
+   * Use default value if limit is not set. And do not allow to exceed the hard limit 100.
    */
   private int parseLimit(int limit) {
-    return (limit <=0 || limit > query_limit) ? query_limit : limit;
+    if (limit <= 0) {
+      return query_limit;
```
Comment (thomasdelhomenie) [inline calendar-webservice/src/main/java/org/exoplatform/calendar/ws/CalendarRestApi.java]: I would rename this variable to `defaultLimit` to make it more clear (the configuration parameter name should be changed as well IMO). And the hard limit would be called `hardLimit`.

File: calendar-webservice/src/main/java/org/exoplatform/calendar/ws/CalendarRestApi.java [hunk 3]
```diff
@@ -189,10 +192,13 @@ public CalendarRestApi(OrganizationService orgService, IdentityManager identityM
     
     int maxAge = 604800;
     if (params != null) {
-      if (params.getValueParam("query_limit") != null) {
-        query_limit = Integer.parseInt(params.getValueParam("query_limit").getValue());        
+      if (params.getValueParam("default.limit") != null) {
+        defaultLimit = Integer.parseInt(params.getValueParam("default.limit").getValue());
       }
-      
+      if (params.getValueParam("default.limit") != null) {
```
Comment (thomasdelhomenie) [inline calendar-webservice/src/main/java/org/exoplatform/calendar/ws/CalendarRestApi.java]: Should be "hard.limit" here, right ?

Comment (trongtt) [reply calendar-webservice/src/main/java/org/exoplatform/calendar/ws/CalendarRestApi.java]: Yes, indeed.

**Pull request merged**